### PR TITLE
Add rate limit to changefeed error, refine owner rate limit

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Owner could be restarted infinitely due to some changefeed error, such as `newChangeFeed` always fails. We should add more rate limit guard to owner.

### What is changed and how it works?

- Add error timestamp record in a new `ErrorHis` field in `ChangeFeedInfo`, to pass this data across different captures (as owner can change).
- Delay to start a changefeed if it fails too frequent.
- Add rate limit before a capture campaigns to be owner.
- Re-enable two test cases in `owner_test.go`, update some logic according to current code

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release note

- No release note

### CODE REVIEW

File: cdc/model/changefeed.go [hunk 1]
```diff
@@ -58,9 +72,10 @@ type ChangeFeedInfo struct {
 	Engine       SortEngine   `json:"sort-engine"`
 	SortDir      string       `json:"sort-dir"`
 
-	Config *config.ReplicaConfig `json:"config"`
-	State  FeedState             `json:"state"`
-	Error  *RunningError         `json:"error"`
+	Config   *config.ReplicaConfig `json:"config"`
+	State    FeedState             `json:"state"`
+	ErrorHis []int64               `json:"history"`
```
Comment (leoppro) [inline cdc/model/changefeed.go:77]: how about using `time.Duration` instead of `int64`?

Comment (amyangfei) [reply cdc/model/changefeed.go:77]: This filed will be stored into etcd, and may be shared by different captures (as the owner can be transferred), so int64 is used here to store timestamp.

File: cdc/model/changefeed.go [hunk 2]
```diff
@@ -44,6 +45,19 @@ const (
 	StateRemoved FeedState = "removed"
 )
 
+const (
+	// ErrorHistoryGCInterval represents how long we keep error record in changefeed info
+	ErrorHistoryGCInterval = time.Minute * 10
```
Comment (overvenus) [inline cdc/model/changefeed.go:50]: Looks like them can be unexported.

**Pull request merged**